mono/mini/mini.h

```diff
@@ -2363,6 +2363,53 @@ mono_bb_last_inst (MonoBasicBlock *bb, int filter)
 	return ins;
 }
 
+
+typedef struct _MonoInterpStackIter MonoInterpStackIter;
+
+/* Needed for stack allocation */
+struct _MonoInterpStackIter {
+	gpointer dummy [8];
+};
+
+typedef gpointer MonoInterpFrameHandle;
+
+int
+mono_interp_regression_list (int verbose, int count, char *images []);
+
+void
+mono_interp_init (void);
+
+void
+mono_interp_stub_init (void);
+
+typedef struct {
+	gpointer (*create_method_pointer) (MonoMethod *method, MonoError *error);
+	MonoObject* (*runtime_invoke) (MonoMethod *method, void *obj, void **params, MonoObject **exc, MonoError *error);
+	void (*init_delegate) (MonoDelegate *del);
+#ifndef DISABLE_REMOTING
+	gpointer (*get_remoting_invoke) (gpointer imethod, MonoError *error);
+#endif
+	gpointer (*create_trampoline) (MonoDomain *domain, MonoMethod *method, MonoError *error);
+	void (*walk_stack_with_ctx) (MonoInternalStackWalk func, MonoContext *ctx, MonoUnwindOptions options, void *user_data);
+	void (*set_resume_state) (MonoJitTlsData *jit_tls, MonoException *ex, MonoJitExceptionInfo *ei, MonoInterpFrameHandle interp_frame, gpointer handler_ip);
+	gboolean (*run_finally) (StackFrameInfo *frame, int clause_index, gpointer handler_ip);
+	gboolean (*run_filter) (StackFrameInfo *frame, MonoException *ex, int clause_index, gpointer handler_ip);
+	void (*frame_iter_init) (MonoInterpStackIter *iter, gpointer interp_exit_data);
+	gboolean (*frame_iter_next) (MonoInterpStackIter *iter, StackFrameInfo *frame);
+	MonoJitInfo* (*find_jit_info) (MonoDomain *domain, MonoMethod *method);
+	void (*set_breakpoint) (MonoJitInfo *jinfo, gpointer ip);
+	void (*clear_breakpoint) (MonoJitInfo *jinfo, gpointer ip);
+	MonoJitInfo* (*frame_get_jit_info) (MonoInterpFrameHandle frame);
+	gpointer (*frame_get_ip) (MonoInterpFrameHandle frame);
+	gpointer (*frame_get_arg) (MonoInterpFrameHandle frame, int pos);
+	gpointer (*frame_get_local) (MonoInterpFrameHandle frame, int pos);
+	gpointer (*frame_get_this) (MonoInterpFrameHandle frame);
+	MonoInterpFrameHandle (*frame_get_parent) (MonoInterpFrameHandle frame);
+	void (*start_single_stepping) (void);
+	void (*stop_single_stepping) (void);
+} MonoInterpCallbacks;
+
```

Comment:
This could go to interp.h with a forward declaration, i.e.:
struct _MonoInterpCallbacks;

typedef struct _MonoInterpCallbacks MonoInterpCallbacks;

and in interp.h:

struct _MonoInterpCallbacks {

Reply:
good point! I changed that and rebased this PR; I had to do a couple changes due to https://github.com/mono/mono/pull/6072 being merged. please review again 🙂